src/main/java/producer/RecordBatch.java

```diff
@@ -0,0 +1,62 @@
+package producer;
+
+import org.tinylog.Logger;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class RecordBatch {
+    private final int maxBatchSizeInBytes;
+    private int currBatchSize;
+    private int numRecords;
+    private List<byte[]> batch;
+
+    public RecordBatch() {
+        this(10_000); // default batch size: 10 KB = 100,00 bytes
+    }
+
+    public RecordBatch(int maxBatchSizeInBytes) {
+        this.maxBatchSizeInBytes = maxBatchSizeInBytes;
+        this.currBatchSize = 0;
+        this.numRecords = 0;
+        this.batch = new ArrayList<>();
+    }
+
+    // NOTE: This method may need refactoring once SerializedRecord (Producer) is implemented, but logic remains same
+    public boolean append(byte[] serializedRecord) {
+        // use a boolean to represent if a record could fit in the current batch or not
+        // if not, use the returned `false` value as a signal to create an additional batch
+        if (currBatchSize + serializedRecord.length > maxBatchSizeInBytes) {
+            Logger.warn("Record can not in current batch. New one may be necessary");
```

Comment:
OOC, is this log line supposed to say "Record can not fit in current batch. New one may be necessary"?

Reply:
ya